lib/CL/pocl_llvm_api.cc

```diff
@@ -231,21 +232,48 @@ static void get_build_log(cl_program program,
 int pocl_llvm_build_program(cl_program program, 
                             unsigned device_i,
                             const char* user_options_cstr,
-                            char* program_bc_path)
+                            char* program_bc_path,
+                            cl_uint num_input_headers,
+                            const cl_program *input_headers,
+                            const char **header_include_names)
 
 {
   void* write_lock = NULL;
   char tempfile[POCL_FILENAME_LENGTH];
   tempfile[0] = 0;
   llvm::Module **mod = NULL;
+  char temp_include_dir[POCL_FILENAME_LENGTH];
   std::string user_options(user_options_cstr ? user_options_cstr : "");
   std::string content;
   llvm::raw_string_ostream sos(content);
   size_t n = 0;
+  int error;
 
   llvm::MutexGuard lockHolder(kernelCompilerLock);
   InitializeLLVM();
 
+  if (num_input_headers > 0) {
+    error = pocl_cache_create_tempdir(temp_include_dir);
+    std::string tempdir(temp_include_dir);
+    assert(error == 0);
+
+    for (n = 0; n < num_input_headers; n++) {
+      char *input_header = input_headers[n]->source;
+      size_t input_header_size = strlen(input_header);
+      const char *header_name = header_include_names[n];
+      std::string header(header_name);
+      /* TODO this path stuff should be in utils */
```

Comment:
Yes. And somehow more windows-compatible. That is, abstact the directory separator.